### DISCUSSION TITLE
Use already-embeded font to fill PDF form

### DISCUSSION THREAD

**Discussion by flavianh** [Hopding/pdf-lib#1093] [Q&A]
Hello, I have a PDF with some pre-existing embeded fonts. I have to first say that I don't know the PDF standard very well but I'm trying to guess how it behaves.
- When I fill the form with Google Chrome the text is well-aligned, at the right size and the right font.
- When I fill it with the Preview app on macos, the text is filled in the (probably) default font
- When I fill it with `pdf-lib`, the behavior is the same as with Preview: default font. I haven't done the [font manipulation](https://github.com/Hopding/pdf-lib#creating-and-filling-forms) technique from the docs as the fonts are already embeded. 

From this I suspect that:
- There is a possibility to know the font of a PDFTextField
- Google Chrome does it
- `pdf-lib` does not support it

Is this a missing feature or am I misunderstanding the situation?

Cheers

**Comment by glimmbo** [comment]
You'll want to look at the PDFTextField.acroField.DA:
```js
  textField.acroField.DA()
  textField.acroField.setDefaultAppearance(`<DA string>`)
```

**Comment by flavianh** [comment]
Have you actually tried it? Because I've solved this problem in the meantime and I'm 100% confident this is not the solution.
The solution is to actually do this:

```typescript
const file = 'path-to-pdf-file.pdf'
const pdfDoc = await PDFDocument.load(fs.readFileSync(file));
pdfDoc.registerFontkit(fontkit);

// This can be guessed or known by looking at all fields of type fonts
const refNumber = 111;
const embededFontRef = pdfDoc.context.lookup(
PDFRef.of(refNumber, 0),
PDFStream,
);

// Some fonts need to be inflated, others not, to yield the font bytes
const fontBytes = pako.inflate((embededFontRef as PDFRawStream).contents);
const embededFont = await pdfDoc.embedFont(fontBytes);
const form = pdfDoc.getForm();

const field = form.getTextField('Field');
field.setText('The quick brown dog jumps over the lazy fox');

// You have to updateAppearances with the font so that the DefaultAppearance gets changed
field.updateAppearances(embededFont);

const pdfBytes = await pdfDoc.save();
```
